packages/kbn-gateway/src/kibana/kibana_service.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fetch from 'node-fetch';
+import type { Logger, LoggerFactory } from '@kbn/logging';
+import { ConfigStart } from '../config';
+import { ServerStart } from '../server';
+import type { KibanaConfigType } from './kibana_config';
+
+interface KibanaServiceStartDependencies {
+  server: ServerStart;
+}
+
+interface KibanaServiceDependencies {
+  logger: LoggerFactory;
+  config: ConfigStart;
+}
+
+/**
+ * A service to interact with the configured `kibana.hosts`.
+ */
+export class KibanaService {
+  private readonly log: Logger;
+  private readonly config: ConfigStart;
+
+  constructor({ logger, config }: KibanaServiceDependencies) {
+    this.log = logger.get('kibana-service');
+    this.config = config;
+  }
+
+  async start({ server }: KibanaServiceStartDependencies) {
+    server.addRoute({
+      method: 'GET',
+      path: '/api/status',
+      handler: async (req: any, h: any) => {
+        const responses = await this.fetchKibanaResponses('/api/status');
+        // For now we're being super naïve and returning the highest status code
+        const statusCode = responses.reduce((acc, cur) => (cur.status > acc ? cur.status : acc), 0);
```

Comment:
NIT: I would extract that (`fetchKibanaResponses` + the 'aggregation' logic living in the handler) to a dedicated file.

Reply:
Yeah that was what I assumed we'd do as a part of https://github.com/elastic/kibana/issues/141885, but I will split it out for now. Will be easier to test anyhow.